thunder/clang/__init__.py

```diff
@@ -715,8 +727,49 @@ def _advanced_indexing(a: TensorLike, /, key) -> TensorLike:
         lambda: f"Advanced indexing currently only supports keys that are ellipses, integer tensors or sequences, but got {key=}",
     )
 
+    def _to_tensorproxies(x: list, device: devices.DeviceType):
```

Comment:
Why do you want to convert a list (or other sequence?) to a tensor?

Reply:
The reason behind this was because, before this PR, advanced indexing supported Tensor indexing. I thought it would be better to use the existing functionality when receiving a list.

Reply:
In order to do the linear indices / reduce to take we would want tensors for tensor arithmetic.
The list or sequence is a bit complicated due to tuple being special in getitem (so `a[[1, 2, 3]]` is the same as `a[((1, 2, 3),)]` but not as `a[(1, 2, 3)]` which is `a[1, 2, 3]`). For writing things, list clearly using a list is *the* correct thing.

Reply:
The thing I worry about with converting a list into a tensor is that the list could be constexpr (known at compile-time) while we don't really have a notion of tensors with constexpr data (although we could do that, too, I suppose).

Reply:
Do you think it'll be helpful to submit an issue regarding that? Would you be able to provide me an example of the scenario that you are particularly concerned about?

Reply:
No, that's OK -- it's a pretty niche concern we probably won't need to bother with for a long time

Reply:
Ok, cool. Let me resolve this!